spec/app/domain/services/create_course/service_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe Services::CreateCourse::Service do
+  let(:service) { Services::CreateCourse::Service.new }
+  let(:action) { process }
+
+  def process
+    service.process(course_uuid: given_course_uuid, ecosystem_uuid: given_ecosystem_uuid)
+  end
+
+  context "when non-existing ecosystem is given" do
+    let(:given_course_uuid) { SecureRandom.uuid.to_s }
+    let(:given_ecosystem_uuid) { SecureRandom.uuid.to_s }
+
+    let!(:split_time) { time = Time.now; sleep(0.001); time }
+
+    it "raises error" do
+      expect{action}.to raise_error(Errors::AppUnprocessableError)
+    end
+  end
+
+  context "when existing ecosystem is given" do
+
+    ecosystem_uuid = SecureRandom.uuid.to_s
+    course_uuid = SecureRandom.uuid.to_s
+
+    let(:given_course_uuid) { course_uuid }
+    let(:given_ecosystem_uuid) { ecosystem_uuid }
+
+    let!(:split_time) { time = Time.now; sleep(0.001); time }
+
+    expected = {created_course_uuid: course_uuid}
+
+    before(:each) do
+      Ecosystem.find_or_create_by(uuid: ecosystem_uuid)
```

Comment:
There is no reason to believe that this `Ecosystem` should exist, so `Ecosystem.create!` is sufficient (we want an exception if it fails so it's easier to debug what's happening).  Using the factory will [mostly] future-proof these specs from column updates: `create(:ecosystem, uuid: ...)`.